src/controls/qml/RemorseTimer.qml

```diff
@@ -0,0 +1,265 @@
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the author nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import QtQuick 2.9
+import org.asteroid.controls 1.0
+
+/*!
+    \qmltype RemorseTimer
+    \inqmlmodule org.asteroid.controls
+    \brief A full-screen timer component for cancellable actions with a visual countdown.
+
+    This component displays a semi-transparent overlay with a segmented arc countdown,
+    an action message (e.g., "Powering Off"), and a "Tap to cancel" prompt. It triggers
+    a specified action after a set interval unless cancelled by tapping the screen.
+    Used for confirming critical actions like power off or reboot.
+
+    The countdown is visualized by a SegmentedArc gauge, with configurable segment count,
+    start angle, and arc length for flexible styling.
+
+    Example usage in a quick settings toggle:
+    \qml
+    import QtQuick 2.9
+    import org.asteroid.controls 1.0
+
+    Item {
+        id: root
+        RemorseTimer {
+            id: remorse
+            action: qsTrId("id-power-off")
+            interval: 3000
+            gaugeSegmentAmount: 6
+            gaugeStartDegree: -130
+            gaugeEndFromStartDegree: 265
+            onTriggered: console.log("Power off executed")
+            onCancelled: console.log("Power off cancelled")
+        }
+
+        MouseArea {
+            anchors.fill: parent
+            onClicked: remorse.start()
+        }
+    }
+    \endqml
+*/
+Rectangle {
+    id: remorseTimer
+    anchors.fill: parent
+    color: Qt.rgba(0, 0, 0, 0.8)
+    visible: false
+    z: 100
+    opacity: 0
+
+    /*!
+        \qmlproperty string RemorseTimer::cancelText
+        The text displayed for the cancel prompt (e.g., "Tap to cancel"). Can be a translated string.
+    */
+    property string cancelText: "Tap to cancel"
+
+    /*!
+        \qmlproperty string RemorseTimer::action
+        The action message displayed (e.g., "Powering Off"). Should be a translated string.
```

Comment:
Why the distinction between `Should be a translated string.` and `Can be a translated string.` between the `action` and `cancelText`?

Either way, is it relevant that both should be translated? This is a detail for whoever decides to use it right?